core/block_validator.go

```diff
@@ -131,6 +132,24 @@ func (v *BlockValidator) ValidateState(block *types.Block, statedb *state.StateD
 	if root := statedb.IntermediateRoot(v.config.IsEIP158(header.Number)); header.Root != root {
 		return fmt.Errorf("invalid merkle root (remote: %x local: %x) dberr: %w", header.Root, root, statedb.Error())
 	}
+	if blockEw := block.ExecutionWitness(); blockEw != nil {
```

Comment:
Finally, we reached the main goal of the PR. `ValidateState` is an existing method that has the usual stateRoot and gasUsage checks when validating blocks.

Now I added logic to also check the execution witness (if the proposed block has one).